src/diff.rs

```diff
@@ -53,23 +60,97 @@ pub struct MoveFile {
     pub new_path: Path,
 }
 
+#[derive(Debug, PartialEq, Eq)]
+pub struct CopyFile {
+    pub old_path: Path,
+    pub new_path: Path,
+}
+
 #[derive(Debug, PartialEq, Eq)]
 pub struct ModifiedFile {
     pub path: Path,
     pub diff: FileDiff,
 }
 
+/// A set of changes belonging to one file.
+#[derive(Debug, PartialEq, Eq)]
+pub enum FileDiff {
+    Binary,
+    Plain(Vec<Hunk>),
+}
+
+/// A set of line changes.
 #[derive(Debug, PartialEq, Eq)]
-pub struct FileDiff {
-    // TODO
+pub struct Hunk {
+    pub header: Line,
+    pub lines: Vec<LineDiff>,
+}
+
+/// The content of a single line.
+pub type Line = Vec<u8>;
+
+/// Single line delta. Two of these are need to represented a modified line: one
+/// addition and one deletion. Context is also represented with this type.
+#[derive(Debug, PartialEq, Eq)]
+pub struct LineDiff {
+    /// Line number in old or new file.
+    pub line_num: u32,
+    /// Line content.
+    pub line: Line,
+    /// Line diff kind, eg. addition or deletion.
+    pub kind: LineDiffKind,
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum LineDiffKind {
```

Comment:
Could you document these? I'm adding a ci check on another branch to run `cargo doc --no-deps` so it'd be good to anticipate that :)